resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -59,7 +61,24 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
-          PodRunning(pod)
+          // TODO: upcoming Kubernenetes feature will make this code redundant
+          // https://github.com/kubernetes/enhancements/issues/753
+          // Checking executor container status is not terminated
+          // Pod status can still be running if sidecar container status is running
+          val executorContainerStatusCode = pod.getStatus.getContainerStatuses.asScala.
+            filter(_.getName == DEFAULT_EXECUTOR_CONTAINER_NAME).
```

Comment:
Same comment as above. If the container name is generally enforced elsehwere that's ok but if we're depending on Spark being the only one to launch we should see if there is another filter we could do (maybe a tag)?

Reply:
I think you made a really good point, but as far as I know, there is no identifier of container expect the container name. maybe we have to force people use this default name.

Reply:
Ok, this is probably a rare case but (for example) the Google K8s Spark folks launch their own containers I believe, so let's try and make not of this in the migration guide.

Reply:
@holdenk @jinxingwang one option is to introduce a conf option for the executor container name so it can be tracked in cluster mode in case it is modified by the pod template feature.

Reply:
That would be a cool, I think we can probably just mention it in the migration guide if we don't, but if we add setting for configuring the container name I'm good with that too.